### DISCUSSION TITLE
How are layers with dependencies suppose to work?

### DISCUSSION THREAD

**Discussion by JiProchazka** [nuxt/nuxt#26187] [Questions]
Hi,
I have a service-layer with stripe module. In the application called `layer-tester-app` I use the service layer (extend it with service layer). Do I need to install the strip module npm package to the application as well? And do I need to register the stripe module in the service layer and in the application as well?
Because I'm getting:
```
Error while requiring module @unlok-co/nuxt-stripe: Error: Cannot find module '/Users/mymac/Projects/layer-tester-app/@unlok-co/nuxt-stripe'
Require stack:
- /Users/mymac/Projects/layer-tester-app/index.js


 ERROR  Cannot find module '/Users/mymac/Projects/layer-tester-app/@unlok-co/nuxt-stripe'
Require stack:
- /Users/mymac/Projects/layer-tester-app/index.js
```

My service nuxt.config.ts looks like this:
```typescript
export default defineNuxtConfig({
  devtools: { enabled: true },
  modules: ["@unlok-co/nuxt-stripe"],
  stripe: {
    server: {
      key: "",
    },
    client: {
      key: "",
    },
  },
});
```
And my layer-tester-app nuxt.config.ts is:
```typescript
export default defineNuxtConfig({
  devtools: { enabled: true },
  extends: "../common-layer",
});
```

Those two apps are aside in the folder structure:
-Projects
  --layer-tester-app
  --service-layer

Thanks

**Comment by AloisSeckar** [comment]
one think to have in mind - external dependencies should be declared in `dependencies` and not `devDependencies` of `package.json` in the base layer

and maybe you should try:
```
extends: [
  ["../common-layer", { install: true }]
]
```
although I am not sure, if it works the same for extending local layers (this syntax is for layers as npm packages). 

But generally you should be able to inherit all deps declared and modules installed in base layer without having to explictly re-declare them.

**Reply by JiProchazka** [reply]
Ok...
I have tried a completely new empty nuxt app extended by a public github repo and I'm getting exactly the same...

```typescript
export default defineNuxtConfig({
  devtools: { enabled: true },
  extends: [
    ["github:MyPublicGit/service-layer", { install: true }]
  ],
});
```

Can the problem be, that I have created the layer from the https://nuxt.new/ layer started template? Isn't it outdated?

**Comment by v0id-4lpz** [comment]
Hi!
Same problem from days and can't find a working solution... I'm stuck!

Linked my layer (github) to a fresh empty Nuxt project and...

```sh
$ npm install

> postinstall
> nuxt prepare


> mylayer@0.0.1 postinstall  
> nuxt prepare .playground

 ERROR  Cannot find module 'vite-plugin-vuetify'                                                                                                                                                                                                                                                           21:20:38
Require stack:
- D:\Projects\myProject\node_modules\.c12\github_xxx\nuxt.config.ts

  Require stack:
  - nuxt.config.ts
  at Module._resolveFilename (node:internal/modules/cjs/loader:1144:15)
  at Function.resolve (node:internal/modules/helpers:187:19)
  at _resolve (D:\Projects\myProject\node_modules\jiti\dist\jiti.js:1:251148)
  at jiti (D:\Projects\myProject\node_modules\jiti\dist\jiti.js:1:253746)
  at D:/Projects/myProject/node_modules/.c12/github_xxx/nuxt.config.ts:1:212
  at evalModule (D:\Projects\myProject\node_modules\jiti\dist\jiti.js:1:256443)
  at Object.jiti (D:\Projects\myProject\node_modules\jiti\dist\jiti.js:1:254371)
  at resolveConfig (/D:/Projects/myProject/node_modules/c12/dist/index.mjs:330:26)
  at extendConfig (/D:/Projects/myProject/node_modules/c12/dist/index.mjs:241:27)
  at loadConfig (/D:/Projects/myProject/node_modules/c12/dist/index.mjs:176:11)
```

Nuxt don't install layer's dependencies or expose them to the main app

**Comment by v0id-4lpz** [comment]
Is someone has a workaround?
I trited some tricks but none works except manually install layer's deps in all of my projects but it's not that I need.
It's very annoying, i'm stuck with that for my work.

**Comment by benjamindedonder** [comment]
I'm experiencing the same issue! I'm extending from a local layer and I installed the layer packages as dependencies and not devDependencies.
